lib/evmone/instructions.hpp

```diff
@@ -923,6 +923,22 @@ inline Result mcopy(StackTop stack, int64_t gas_left, ExecutionState& state) noe
     return {EVMC_SUCCESS, gas_left};
 }
 
+inline Result dataload(StackTop stack, int64_t gas_left, ExecutionState& state) noexcept
+{
+    auto& index = stack.top();
+
+    if (state.data.size() < 32 || (state.data.size() - 32) < index)
+        return {EVMC_INVALID_MEMORY_ACCESS, gas_left};  // TODO: Introduce EVMC_INVALID_DATA_ACCESS
```

Comment:
Not everyone is happy about "invalid memory access" in _return data_. Do we want to have it here?

Reply:
I mean general design: maybe padding with zero is better. But I think this is good staring point.